### PR TITLE
Added snippets for true and false operators/literals

### PR DESCRIPTION
**Pull request by pkulikov**
Supports dotnet/docs#9356

### CODE REVIEW

File: snippets/csharp/keywords/TrueFalseOperatorsExample.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public static bool operator false(LaunchStatus x) => x == Red;
+
+    public static LaunchStatus operator &(LaunchStatus x, LaunchStatus y)
+    {
+        if (x == Red || y == Red || (x == Yellow && y == Yellow))
+        {
+            return Red;
+        }
+
+        return Green;
```
Comment (BillWagner) [inline snippets/csharp/keywords/TrueFalseOperatorsExample.cs]: I think there's an issue here. For `x & y & z` work correctly when `z` is `Yellow`, this should return `Yellow` when one of `x` or `y` is `Yellow`.  

Do you agree?

Comment (pkulikov) [reply snippets/csharp/keywords/TrueFalseOperatorsExample.cs]: @BillWagner 
> For `x & y & z` work correctly...

to clarify: by "correctly" you mean that the `&` overload must be associative, so `(x & y) & z == x & (y & z)`

Yes, I agree then. Will make an update soon.

Comment (pkulikov) [reply snippets/csharp/keywords/TrueFalseOperatorsExample.cs]: @BillWagner I've addressed your feedback; thanks for review.

I've checked the updated definition for all possible `x`, `y`, and `z` values: now the `&` operator is commutative and associative. In particular, that means that `x1 & ... & xN` is `Yellow` iff there is only one operand which is `Yellow` with all others being `Green`. That's why I've updated the `true` definition to return `true` value for `Yellow` operand. The following requirement is met:
> if only one sub-system is yellow, we can launch.

**Comment by BillWagner** [comment]
Thanks for making these updates. I'll :shipit: now and then check the article PR.

**Pull request merged**